providers/implementations/asymciphers/rsa_enc.c

```diff
@@ -353,8 +389,32 @@ static int rsa_set_ctx_params(void *vprsactx, const OSSL_PARAM params[])
 
     p = OSSL_PARAM_locate_const(params, OSSL_ASYM_CIPHER_PARAM_PAD_MODE);
     if (p != NULL) {
-        if (!OSSL_PARAM_get_int(p, &pad_mode))
+        int pad_mode = 0;
+
+        switch (p->data_type) {
+        case OSSL_PARAM_INTEGER: /* Support for legacy pad mode number */
+            if (!OSSL_PARAM_get_int(p, &pad_mode))
+                return 0;
+            break;
+        case OSSL_PARAM_UTF8_STRING:
```

Comment:
I'm not keen on this "dual type" solution at all. This makes it difficult for third party provider authors to provide a drop-in replacement. They may have to support the "legacy" way of doing things too - otherwise it may not work for all applications.

Reply:
Yeah well, I do not know how otherwise. Are you saying that we should translate numbers to strings in libcrypto, in whatever'se the name of the specialised setter and getter functions? It is kinda odd, considering that this is really entirely in the backend (in this regard, the ameth and pmeth files are equal to providers)

Reply:
The thing is that, even with legacy, it depends on what backend is used. Could be an engine...

Reply:
> Are you saying that we should translate numbers to strings in libcrypto, in whatever'se the name of the specialised setter and getter functions? 

I think I'd prefer that since its legacy only and will eventually go away. It keeps things simpler for provider authors.

(BTW, should we deprecate EVP_PKEY_CTX_ctrl, EVP_MD_CTX_ctrl, etc?...not for htis PR)